### PR TITLE
feat: refactor tsconfig smart contracts

### PR DESCRIPTION
**Pull request by r0x4xn**
https://www.notion.so/Refactor-tsconfig-for-smart-contract-applications-on-the-monorepo-bac563c53c2b4abc88a5d31cb70236e2

### CODE REVIEW

**Review by XavierEkkel — APPROVED** [review]
LGTM after questions

File: packages/tsconfig/smart-contracts.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "https://json.schemastore.org/tsconfig",
+  "display": "Server",
```
[the remark this answers is not in the document]
Comment (r0x4xn) [reply packages/tsconfig/smart-contracts.json:3]: The name of the configuration. This should be changed to **Smart Contracts**. Will change 👍

File: packages/tsconfig/smart-contracts.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "exclude": ["node_modules", ".turbo"],
+  "extends": "./base.json",
+  "include": ["**/*"],
+  "compilerOptions": {
+    "target": "ES2019",
+    "lib": ["ES2019"],
+    "noEmit": false,
+    "module": "commonjs",
```
[the remark this answers is not in the document]
Comment (r0x4xn) [reply packages/tsconfig/smart-contracts.json:11]: capitalization doesn't matter

File: packages/tsconfig/smart-contracts.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "https://json.schemastore.org/tsconfig",
+  "display": "Server",
+  "exclude": ["node_modules", ".turbo"],
+  "extends": "./base.json",
+  "include": ["**/*"],
+  "compilerOptions": {
+    "target": "ES2019",
+    "lib": ["ES2019"],
+    "noEmit": false,
```
[the remark this answers is not in the document]
Comment (r0x4xn) [reply packages/tsconfig/smart-contracts.json:10]: Taken from the `token` configuration project. If we have it there, we can have it in all for consistency

File: packages/tsconfig/smart-contracts.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    "module": "commonjs",
+    "sourceMap": true,
+    "isolatedModules": false,
```
[the remark this answers is not in the document]
Comment (r0x4xn) [reply packages/tsconfig/smart-contracts.json:13]: Taken from the `token` configuration project. If we have it there, we can have it in all for consistency

**Comment by r0x4xn** [comment]
App <strong>docs</strong> has been deployed to Vercel.

<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://docs-6eft4wd8f-prepo.vercel.app'>https://docs-6eft4wd8f-prepo.vercel.app</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="prj_P9R3CtLzShfhpRDRTnvFqguGIvtB"></hidden>

**Comment by r0x4xn** [comment]
App <strong>simulator</strong> has been deployed to Vercel.

<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://simulator-jjytqqurv-prepo.vercel.app'>https://simulator-jjytqqurv-prepo.vercel.app</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="prj_JqVcxQPRfkDbLp3ytm7FLLIvOPXc"></hidden>

File: packages/tsconfig/smart-contracts.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (XavierEkkel) [reply packages/tsconfig/smart-contracts.json:10]: *`server.json`

**Comment by r0x4xn** [comment]
App <strong>core-dapp</strong> has been deployed to Vercel.

<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://core-dapp-rfx9y54kw-prepo.vercel.app'>https://core-dapp-rfx9y54kw-prepo.vercel.app</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="prj_JWVJ3WoV71UcOmB52qrcezjyBL3V"></hidden>

**Comment by r0x4xn** [comment]
App <strong>react-boilerplate</strong> has been deployed to Vercel.

<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://react-boilerplate-q1hh0ruuj-prepo.vercel.app'>https://react-boilerplate-q1hh0ruuj-prepo.vercel.app</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="prj_3Lydru5gdN5RuqWFQBT8AZrbiROq"></hidden>

**Pull request merged**